packages/statemanager/test/stateManager.account.spec.ts

```diff
@@ -0,0 +1,177 @@
+      st.ok(!equalsBytes(stateManager._trie.root(), KECCAK256_RLP), 'it has a new root')
+
+      // set state root to empty trie root
+      await stateManager.setStateRoot(KECCAK256_RLP)
+
+      const res = await stateManager.getStateRoot()
+      st.ok(equalsBytes(res, KECCAK256_RLP), 'it has default root')
+      st.end()
+    })
+
+    t.test('should clear the cache when the state root is set', async (st) => {
+      const stateManager = new DefaultStateManager({ accountCacheOpts })
+      const address = new Address(hexStringToBytes('a94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+      const account = createAccount()
+
+      // test account storage cache
+      const initialStateRoot = await stateManager.getStateRoot()
+      await stateManager.checkpoint()
+      await stateManager.putAccount(address, account)
+
+      const account0 = await stateManager.getAccount(address)
+      st.equal(account0!.balance, account.balance, 'account value is set in the cache')
+
+      await stateManager.commit()
+      const account1 = await stateManager.getAccount(address)
+      st.equal(account1!.balance, account.balance, 'account value is set in the state trie')
+
+      await stateManager.setStateRoot(initialStateRoot)
+      const account2 = await stateManager.getAccount(address)
+      st.equal(account2, undefined, 'account is not present any more in original state root')
+
+      // test contract storage cache
+      await stateManager.checkpoint()
+      const key = hexStringToBytes(
+        '1234567890123456789012345678901234567890123456789012345678901234'
+      )
+      const value = hexStringToBytes('1234')
+      await stateManager.putAccount(address, account)
+      await stateManager.putContractStorage(address, key, value)
+
+      const contract0 = await stateManager.getContractStorage(address, key)
+      st.ok(equalsBytes(contract0, value), "contract key's value is set in the _storageTries cache")
+
+      await stateManager.commit()
+      await stateManager.setStateRoot(initialStateRoot)
+      try {
+        await stateManager.getContractStorage(address, key)
+      } catch (e) {
+        st.pass('should throw if getContractStorage() is called on non existing address')
+      }
+
+      st.end()
+    })
+
+    t.test(
+      'should put and get account, and add to the underlying cache if the account is not found',
+      async (st) => {
+        const stateManager = new DefaultStateManager({ accountCacheOpts })
+        const account = createAccount()
+        const address = new Address(hexStringToBytes('a94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+
+        await stateManager.putAccount(address, account)
+
+        const res1 = await stateManager.getAccount(address)
+
+        st.equal(res1!.balance, BigInt(0xfff384))
+
+        await stateManager.flush()
+        stateManager._accountCache?.clear()
+
+        const res2 = await stateManager.getAccount(address)
+
+        st.ok(equalsBytes(res1!.serialize(), res2!.serialize()))
```

Comment:
This test does not test if the account is added to the cache as far as I can see.

Reply:
No, but this is also not a cache test, so goal here was to test StateManager functionality.